oci/private/authn.bzl

```diff
@@ -202,15 +283,40 @@ def _get_token(rctx, state, registry, repository):
             if url in state["token"]:
                 return state["token"][url]
 
-            rctx.download(
-                url = [url],
-                output = "www-authenticate.json",
-                # optionally, sending the credentials to authenticate using the credentials.
-                # this is for fetching from private repositories that require WWW-Authenticate
-                auth = {url: pattern},
-            )
+            auth = None
+            if pattern["login"] == "<token>":
+                if versions.is_at_least("7.0.0", versions.get() or "0.0.0"):
```

Comment:
even though i like the feature detection here; we probably should just use `rctx.os.environ` for now to match rest of the repo, and add `OCI_ENABLE_OAUTH2_SUPPORT` to list of env that invalidates this repo.